### PR TITLE
Use live-built corehost on s390x for creating testhost

### PR DESCRIPTION
**Pull request by akoeplinger**
This allows consuming the dotnet host that was just built instead of relying on a downloaded one (which doesn't exist for s390x).
We need to move the `host` subset build before the `libs` subset so that the artifacts are available before externals.csproj runs.

### CODE REVIEW

**Comment by akoeplinger** [comment]
@ViktorHofer let me know if this makes sense, especially the subsets move.

**Comment by jkoritzinsky** [comment]
cc: @agocke

File: eng/Subsets.props [hunk 1]
```diff
@@ -120,7 +120,14 @@
...
     <SubsetName Include="Mono.MsCorDbi" Description="The implementation of ICorDebug interface." />
     <SubsetName Include="Mono.Workloads" OnDemand="true" Description="Builds the installers and the insertion metadata for Blazor workloads." />
-    
+
+    <!-- Host -->
+    <SubsetName Include="Host" Description="The .NET hosts, packages, hosting libraries, and tests." />
+    <SubsetName Include="Host.Native" Description="The .NET hosts." />
+    <SubsetName Include="Host.Pkg" Description="The .NET host packages." />
+    <SubsetName Include="Host.Tools" Description="The .NET hosting libraries." />
+    <SubsetName Include="Host.Tests" Description="The .NET hosting tests." />
```
Comment (jkoritzinsky) [inline eng/Subsets.props:129]: Can you move the Host.Tests subset to after the libs subsets? The host tests consume the live libraries assets during test run, so they need to be sequenced after the libraries build.

I know the right place to comment this is lower, but I GitHub won't let me comment on the line.

Comment (akoeplinger) [reply eng/Subsets.props:129]: Yup, done.

**Comment by am11** [comment]
Nice! I thought it is not possible to use live host due to some bindings with SDK (e.g. host changes API that goes hand-in-hand with the SDK), which is why we wait for the whole update cycle: runtime->sdk->{installer,arcade}->runtime.
If it is not the case then why limit it to just s390x; all platforms can take advantage of live host, no?

**Comment by akoeplinger** [comment]
> If it is not the case then why limit it to just s390x; all platforms can take advantage of live host, no?

My understanding is that the main reason for the current setup is so you don't need to have the native toolchain/compilers to build the corehost if you just want to work on the managed libraries and run tests.

**Comment by jkoritzinsky** [comment]
The current issue is more with how we split up our CI pipeline. Right now we build the hosts after the libraries because the host tests require the libraries tests (and because that's how it worked before repo consolidation).

**Comment by ViktorHofer** [comment]
What Jeremy said makes sense but it's also what Alex says. Currently libraries devs are able to just build the libs subset. When building the live host that wouldn't be possible anymore with the current loosely defined dependency. We would want to establish a dependency from the libs test execution to the host.

**Comment by akoeplinger** [comment]
Please let me know if there's anything missing here or if you don't agree with this approach :)

**Review by ViktorHofer — APPROVED** [review]

**Comment by akoeplinger** [comment]
FYI @uweigand @nealef

**Pull request merged**